client/app/components/auth/user.service.js

```diff
@@ -1,46 +1,45 @@
 angular.module('throwcast.auth')
 
-.factory('userService', function ($http) {
+.factory('userService', function ($http, $window, $location) {
   var user;
   var podcastQueue;
   var message;
   return {
+    createUser: function (userCredentials) {
+      return $http.post('http://localhost:8888/api/users/', userCredentials);
+    },
     getUser: function () {
       return $http.get('http://localhost:8888/users/me')
       .then(function (res) {
-        user.data = res.data;
+        user = res.data;
```

Comment:
This reassigns the user variable. Any service that has grabbed the user object from `userService.user` will have a stale reference. You should be changing the properties of user no doing a reassignment.